code/modules/cargo/randomstock.dm

```diff
@@ -0,0 +1,1880 @@
+	"debugger" = 2,
+	"surgerykit" = 2.5,
+	"crimekit" = 1,
+	"carpet" = 2,
+	"gift" = 4,
+	"lightfloor" = 2,
+	"linenbin" = 1,
+	"coatrack" = 1,
+	"riotshield" = 2,
+	"fireaxe" = 1.5,
+	"service" = 2,
+	"robot" = 2,
+	"latexb" = 1.2,
+	"taperoll" = 1,
+	"headset" = 2,
+	"bat" = 1.2,
+	"scythe" = 1,
+	"manual" = 2,
+	"jammer" = 2,
+	"rped" = 2,
+	"briefcase" = 2,
+	"blade" = 1.5,
+	"exoquip" = 2,
+	"laserscalpel" = 1.5,
+	"electropack" = 1,
+	"nothing" = 0)
+
+var/list/global/random_stock_rare = list(
+	"gold" = 2.5,
+	"diamond" = 1.5,
+	"uranium" = 3,
+	"EMP" = 1,
+	"hypercell" = 3,
+	"combatmeds" = 3,
+	"batterer" = 1,
+	"posibrain" = 3,
+	"augmentvision" = 1.6,
+	"thermals" = 2,
+	"bsbeaker" = 3,
+	"energyshield" = 3,
+	"hardsuit" = 0.75,
+	"cluster" = 2.0,
+	"cloak" = 1,
+	"sword" = 1,
+	"ims" = 1.5,
+	"exogear" = 1.5,
+	"teleporter" = 1,
+	"voice" = 2,
+	"nothing" = 0)
+
+var/list/global/random_stock_large = list(
+	"russian" = 1,
+	"emergency" = 2,
+	"firecloset" = 2,
+	"tacticool" = 0.4,
+	"radsuit" = 3,
+	"exosuit" = 1.2,//A randomly generated exosuit in a very variable condition.
+	"EOD"	=	1.5,
+	"biosuit" = 3,
+	"hydrotray" = 3,
+	"oxycanister" = 6,//Cargo should almost always have an oxycanister
+	"oxydispenser" = 5,
+	"bubbleshield" = 2,
+	"chassis" = 2,
+	"watertank" = 2,
+	"fueltank" = 2,
+	"airpump" = 1,
+	"airscrubber" = 1,
+	"generator" = 5,
+	"flasher" = 2,
+	"vendor" = 6,
+	"piano" = 2,
+	"suspension" = 2,
+	"animal" = 2.5,
+	"cablelayer" = 1,
+	"floodlight" = 3,
+	"floorlayer" = 2,
+	"heater" = 1.3,
+	"dispenser" = 2.5,
+	"jukebox" = 1.2,
+	"pipemachine" = 1.7,
+	"nothing" = 0)
+
+
+/proc/spawn_cargo_stock()
+	var/start_time = world.timeofday
+	new /datum/cargospawner()
+	admin_notice("<span class='danger'>Cargo Stock generation completed in [round(0.1*(world.timeofday-start_time),0.1)] seconds.</span>", R_DEBUG)
+
+/datum/cargospawner
+	var/list/containers = list()
+	var/list/tables = list()
+	var/list/full_containers = list()//Used to hold references to crates we filled up
+	var/area/warehouse
+
+	var/list/infest_mobs_minor = list(
+		"/mob/living/simple_animal/mouse" = 1,
+		"/mob/living/simple_animal/lizard" = 0.5,
+		"/mob/living/simple_animal/yithian" = 0.7,
+		"/mob/living/simple_animal/tindalos" = 0.6,
+		"/mob/living/bot/secbot" = 0.1)
+
+	var/list/infest_mobs_moderate = list(
+		"/mob/living/simple_animal/bee" = 1,
+		"/mob/living/simple_animal/hostile/diyaab" = 1,
+		"/mob/living/simple_animal/hostile/viscerator" = 1,
+		"/mob/living/simple_animal/hostile/scarybat" = 1)
+
+	var/list/infest_mobs_severe = list(
+		"/mob/living/simple_animal/hostile/giant_spider/hunter" = 1,
+		"/mob/living/simple_animal/hostile/shantak" = 0.7,
+		"/mob/living/simple_animal/hostile/bear" = 0.5,
+		"/mob/living/simple_animal/hostile/carp" = 1.5,
+		"cratey" = 1
+	)
+
+/datum/cargospawner/New()
+
+	//First lets get the reference to our warehouse
+	for(var/areapath in typesof(/area/quartermaster/storage))
+		warehouse = locate(areapath)
```

Comment:
What if there are multiple warehouses ? - This only spawns loot in one
Afaik, the new map has two